File: server/src/shared/db/schema/event.ts
import { integer, primaryKey, sqliteTable, text } from "drizzle-orm/sqlite-core";
import { genresTable } from "./genre";
import { venuesTable } from "./venue";

export const eventsTable = sqliteTable("event", {
  id: integer("id").primaryKey({ autoIncrement: true }),
  title: text("title").notNull(),
  description: text("description").notNull(),
  coverImageUrl: text("cover_image_url").notNull(),
  fromDate: integer("from_date", { mode: "timestamp" }).notNull(),
  toDate: integer("to_date", { mode: "timestamp" }).notNull(),
  venueID: integer("venue_id").notNull().references(() => venuesTable.id)
})

export const eventsGenresTable = sqliteTable("events_genres", {
  eventID: integer("event_id").notNull().references(() => eventsTable.id),
  genreID: integer("genre_id").notNull().references(() => genresTable.id)
}, (table) => ({
  pk: primaryKey({ columns: [table.eventID, table.genreID] })
}))

File: backend/cmd/cli/event.go
package main

import (
	"context"
	"fmt"

	"github.com/jackc/pgx/v5"
)

func createEventTables(ctx context.Context, tx pgx.Tx) error {
	query := `
	CREATE TABLE IF NOT EXISTS event (
		id INTEGER GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
		title TEXT NOT NULL,
		description TEXT NOT NULL,
		ticket_url TEXT NOT NULL,
		image_url TEXT NOT NULL,
		venue_id INTEGER NOT NULL REFERENCES venue(id)
	);

	CREATE TABLE IF NOT EXISTS concert (
		id INTEGER GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
		from_date TIMESTAMP NOT NULL,
		to_date TIMESTAMP NOT NULL,
		event_id INTEGER NOT NULL REFERENCES event(id),
		artist_id INTEGER NOT NULL REFERENCES artist(id)
	);`

	if _, err := tx.Exec(ctx, query); err != nil {
		return fmt.Errorf("Could not execute query: %v", err)
	}

	return nil
}
